src/plugins/vis_builder/public/visualizations/index.ts

```diff
@@ -6,17 +6,26 @@
 import type { TypeServiceSetup } from '../services/type_service';
 import { createMetricConfig } from './metric';
 import { createTableConfig } from './table';
-import { createHistogramConfig, createLineConfig, createAreaConfig } from './vislib';
+import {
+  createHistogramConfig,
+  createLineConfig,
+  createAreaConfig,
+  createPieConfig,
+} from './vislib';
 
-export function registerDefaultTypes(typeServiceSetup: TypeServiceSetup) {
-  const visualizationTypes = [
+export function registerDefaultTypes(typeServiceSetup: TypeServiceSetup, useVega: boolean) {
+  const defaultVisualizationTypes = [
     createHistogramConfig,
     createLineConfig,
     createAreaConfig,
     createMetricConfig,
     createTableConfig,
   ];
 
+  const visualizationTypes = useVega
+    ? [...defaultVisualizationTypes, createPieConfig]
+    : defaultVisualizationTypes;
+
```

Comment:
why not just
```suggestion
  const visualizationTypes = [
    createHistogramConfig,
    createLineConfig,
    createAreaConfig,
    createMetricConfig,
    createTableConfig,
  ];
  
  if (useVega) visualizationTypes.push(createPieConfig);

```

Reply:
I used a similar one before but I got a type error 
```
Argument of type '() => VisualizationTypeOptions<PieOptionsDefaults>' is not assignable to parameter of type '(() => VisualizationTypeOptions<HistogramOptionsDefaults>) | (() => VisualizationTypeOptions<LineOptionsDefaults>) | (() => VisualizationTypeOptions<...>) | (() => VisualizationTypeOptions<...>) | (() => VisualizationTypeOptions<...>)'.Type '() => VisualizationTypeOptions<PieOptionsDefaults>' is not assignable to type '() => VisualizationTypeOptions<HistogramOptionsDefaults>'.Call signature return types 'VisualizationTypeOptions<PieOptionsDefaults>' and 'VisualizationTypeOptions<HistogramOptionsDefaults>' are incompatible.The types of 'ui.containerConfig.style' are incompatible between these types.Type 'StyleTabConfig<PieOptionsDefaults>' is not assignable to type 'StyleTabConfig<HistogramOptionsDefaults>'.Type 'PieOptionsDefaults' is not assignable to type 'HistogramOptionsDefaults'.ts(2345)
```
It's occurring because the visualizationTypes array is being inferred to have a specific type based on its initial contents, and then trying to add a different type (createPieConfig) to it later will cause the type issue. I will define a type for it.